app/_gateway_entities/keyring.md

```diff
@@ -21,20 +21,103 @@ api_specs:
+
+When operating in cluster mode, keyring material propagates automatically among all nodes in the Kong cluster. Because Kong nodes don't have a notion of direct peer-to-peer communication, the underlying data store serves as a communication channel to transmit messages. When a Kong node starts, it generates an ephemeral RSA key pair. The node’s public keys propagate to all other active nodes in the cluster. When an active node sees a message request for keyring material, it wraps the in-memory keyring material in the presented public key, and transmits the payload back over the central messaging channel provided by the underlying data store. This process allows each node in the cluster to broadcast keyring material to new nodes, without sending key material in plain text over the wire. This model requires that at least one node be running at all times within the cluster; a failure of all nodes requires manually re-importing the keyring to one node during an outage recovery.
+
+## Disaster recovery
+
+To automatically back up your Keyring material, make sure to configure the `keyring_recovery_public_key` parameter to point to your `cert.pem` file. See [Enable Keyring](#enable-keyring) for more details.
+
+The Keyring material is then encrypted with the public RSA key defined with the `keyring_recovery_public_key` value in the database. The corresponding private key can be used to decrypt the keyring material in the database:
+```sh
+curl -X POST localhost:8001/keyring/recover -F recovery_private_key=@/path/to/generated/key.pem
+```
+
+The response contains a list of keys that were successfully recovered and a list of keys that could not be recovered. The Kong error log will contain the detailed reason why the keys could not be recovered.
+
+## Enable Keyring
+
+### 1. Generate an RSA key pair
+
+Use the `openssl` CLI to generate an RSA key pair that can be used to export and recover keyring material.
+```sh
+openssl genrsa -out key.pem 2048
+openssl rsa -in key.pem -pubout -out cert.pem
+```
+
+### 2. Configure Kong
+Enabling data encryption in Kong requires modifying the Kong configuration. You can either update `kong.conf` or use environment variables:
+
+{% navtabs %}
+{% navtab "kong.conf" %}
+Set the following values in `kong.conf`:
+```sh
+keyring_enabled = on
+keyring_strategy = cluster
+keyring_recovery_public_key = /path/to/generated/cert.pem
+```
+{% endnavtab %}
+{% navtab "Environment variables" %}
+Create the following environment variables:
+```sh
+export KONG_KEYRING_ENABLED=on
+export KONG_KEYRING_STRATEGY=cluster
+export KONG_KEYRING_RECOVERY_PUBLIC_KEY=/path/to/generated/cert.pem
+```
+{% endnavtab %}
+{% endnavtabs %}
+
+### 3. Start Kong and validate
+Once the configuration is updated, you can start your {{site.base_gateway}} instance and use the following request to make sure the Keyring is enabled:
+
+```sh
+curl -s localhost:8001/keyring
```

Comment:
I tested with the environment variable flow and it worked. I tried doing the kong.conf flow on a new Gateway quickstart install, and for some reason, I kept getting  `{"ids":{}}` before I was able to add the values to kong.conf.

Reply:
Ah you're right, I'm getting the same. After a bit of testing, it looks like the only way to really know if it's working is to look directly in the database to see if the data is encrypted. I'm going to update this part.